llama-index-core/llama_index/core/workflow/events.py

```diff
@@ -10,28 +10,55 @@ class Config:
         arbitrary_types_allowed = True
 
 
-class StartEvent(Event):
-    """StartEvent is implicitly sent when a workflow runs. Mimics the interface of a dict."""
+class DictLikeEvent(Event):
+    """Base class for event types that mimics the interface of a dict."""
 
     _data: Dict[str, Any] = PrivateAttr(default_factory=dict)
 
-    def __init__(self, **data: Any):
-        super().__init__()
+    class Config:
+        arbitrary_types_allowed = True
+
+    def __init__(self, **params: Any):
+        """__init__.
+
+        NOTE: fields and private_attrs are pulled from params by name.
+        """
+        # pull out params for fields
+        fields = {k: v for k, v in params.items() if k in self.__fields__}
+        super().__init__(**fields)
+
+        # set private attrs
+        private_attrs = {
+            k: v for k, v in params.items() if k in self.__private_attributes__
+        }
+        for private_attr, value in private_attrs.items():
+            super().__setattr__(private_attr, value)
```

Comment:
nit: I think we can merge these two `for` loops into one:
```
for k,v in params.items():
    if k in self.__private__attributes__:
        super().__setattr__(k, v)
```

Reply:
Good call. I have to modify this slightly, since we need to ensure that we call `super().__init__()` first before setting any private attributes (order didn't matter I think for Pydantic V1 but it does for Pydantic V2).